module/otp.py

```diff
@@ -0,0 +1,95 @@
+"""
+    OTP module.
+"""
+import os
+import random
+import smtplib
+from email.mime.text import MIMEText
+
+from module.data import MAIL_TEMPLATE
+
+#: Define the following in settings.yaml
+HOST = 'HOST'
+USER = "USER"
+PASSWORD = "PASSWORD"
+
+class OTPException(Exception):
+    pass
+
+class OTPConnectionFailed(OTPException):
+    pass
+
+class OTPLoginFailed(OTPException):
+    pass
+
+class OTPNotAuthenticated(OTPException):
+    pass
+
+class OTPCodeNotSent(OTPException):
+    pass
+
+class OTPSender:
+    """
+        OTP sender class.
+
+        Attributes:
+            authenticated (bool): True if the user is authenticated, False otherwise.
+
+        Methods:
+            login() -> None: 
+                Logs in to the SMTP server. Modifies the authenticated attribute.
+            send(to: str) -> str | None:
+                Sends a randomly generated OTP to the specified email address.
+                Returns the OTP sent to the specified email address.
+    """
+    def __init__(self):
+        try:
+            self.__server = smtplib.SMTP_SSL(HOST, smtplib.SMTP_SSL_PORT)
+        except Exception as e:
+            raise OTPConnectionFailed('Connection failed.') from e
+
+        self.authenticated = False
+        self.login()
+
+    def login(self) -> None:
+        """
+            Logs in to the SMTP server.
+
+            Returns:
+                None: check self.authenticated for login status.
```

Comment:
There is no return value. Just put everything in the description of the function.

You could add a `Raises` [section](https://google.github.io/styleguide/pyguide.html#doc-function-raises) if you want to be precise.
The same comment applies for most functions